examples/init-scripts/haas-network-el6

```diff
@@ -0,0 +1,217 @@
+    lockfile=${LOCKFILE-/var/lock/subsys/$service}
+
+    ## set or override platform specific functions
+    start_daemon() {
+        daemon --user $1 --pidfile $2 "$3 $4"
+    }
+    log_daemon_msg() {
+        echo -n $"$1"
+    }
+    echo_ok() {
+        echo_success; echo
+    }
+    echo_fail() {
+        echo_failure; echo
+    }
+    log_success_msg() {
+        success $"$@"
+    }
+    log_failure_msg() {
+        failure $"$@"
+        echo $"$@"
+    }
+    log_action_msg() {
+        echo $@
+    }
+fi
+
+cd /var/lib/haas
+
+exec=/usr/bin/haas
+
+pidfile=$PID_DIR/haas-network.pid
+
+
+options="serve_networks &"
+
+ensure_dir() {
+    if [ ! -d $1 ]; then
+        mkdir -p $1
+        chown -R $2 $1
+        chmod 755 $1
+    fi
+}
+
+
+start() {
+    log_daemon_msg "Starting $service"
+
+    [ -x $exec ] || exit 5
+
+    status &> /dev/null
+
+    if [ $? -eq 0 ]; then
+        log_action_msg "$service is already running."
+        echo_ok
+        exit 0
+    fi
+
+    ensure_dir $PID_DIR $USER
+
+    start_daemon $USER $pidfile $exec "$options"
+
+    retval=$?
+    sleep 1
+
+    # make sure it's still running. some errors occur only after startup
+    status &> /dev/null
+    if [ $? -ne 0 ]; then
+        echo_fail
+        exit 1
+    fi
+
+    if [ $retval -eq 0 ]; then
+        touch $lockfile
+    fi
+    echo_ok
+
+    return $retval
+}
+
+stop() {
+    log_daemon_msg "Stopping $service"
+
+    if [ -f $pidfile ]; then
+        killproc -p $pidfile $service
+        retval=$?
+        if [ $retval -eq 0 ]; then
+            rm -f $pidfile
+            rm -f $lockfile
+            echo_ok
+        else
+            echo_fail
+        fi
+        sleep 1
+        return $retval
+    else
+        pid=$(pgrep -P1 -fl $exec | grep -v grep | grep -v bash | cut -f1 -d" ")
+        if [ ! "$pid" == "" ]; then
+            kill $pid
+            retval=$?
+            if [ $retval -eq 0 ]; then
+                rm -f $pidfile
+                rm -f $lockfile
+                echo_ok
+            else
+                echo_fail
+            fi
+            sleep 1
+            return $retval
+        else
+            echo_fail
+        fi
+    fi
+}
+
+status() {
+    local pid
+
+    # First try "ps"
+    pid=$(pgrep -P1 -fl $exec | grep -v grep | grep -v bash | cut -f1 -d" ")
+    if [ -n "$pid" ]; then
+        log_action_msg "$service (pid $pid) is running"
+        return 0
+    fi
+
+    # Next try "/var/run/*.pid" files
+    if [ -f "$pidfile" ] ; then
+        read pid < "$pidfile"
+        if [ -n "$pid" ]; then
+            log_action_msg "$service dead but pid file exists"
+            return 1
+        fi
+    fi
+
+    # See if $lockfile
+    if [ -f "$lockfile" ]; then
+        log_action_msg "$service dead but subsys locked"
+        return 2
+    fi
+
+    log_action_msg "$service is stopped"
+    return 3
+}
+
+# sometimes sensu components need a moment to shutdown, so
```

Comment:
I'm willing to bet I know what you used as a reference... I'll give the script a more thourough review when I get the chance, but for now, would you double check that there aren't any more vestigial bits left over from the sensu script? We should avoid having logic in here that only make sense for other daemons.